packages/tests/tests/filter/subscribe.node.spec.ts

```diff
@@ -485,6 +486,52 @@ const runTests = (strictCheckNodes: boolean): void => {
         expectedPubsubTopic: TestPubsubTopic
       });
     });
+
+    it("Renews subscription after lossing a connection", async function () {
+      // setup check
+      expect(waku.libp2p.getConnections()).has.length(3);
+
+      await waku.filter.subscribe(
+        [TestDecoder],
+        serviceNodes.messageCollector.callback
+      );
+
+      await waku.lightPush.send(TestEncoder, messagePayload);
+
+      expect(await serviceNodes.messageCollector.waitForMessages(1)).to.eq(
+        true
+      );
+      serviceNodes.messageCollector.verifyReceivedMessage(0, {
+        expectedMessageText: messageText,
+        expectedContentTopic: TestContentTopic
+      });
+
+      await serviceNodes.confirmMessageLength(1);
+
+      // check renew logic
+      const nwakuPeers = await Promise.all(
+        serviceNodes.nodes.map((v) => v.getMultiaddrWithId())
+      );
+      await Promise.all(nwakuPeers.map((v) => waku.libp2p.hangUp(v)));
+
+      expect(waku.libp2p.getConnections().length).eq(0);
+
+      await Promise.all(nwakuPeers.map((v) => waku.libp2p.dial(v)));
+      await waitForConnections(nwakuPeers.length, waku);
```

Comment:
this is a totally valid point - but I don't think we can check that due to 2 reasons:
- in node js env we deliberately ignore browser API as per https://github.com/waku-org/js-waku/pull/2049#discussion_r1724053578
- even if we consider using a mock for window in test environment AND make it seems like it is offline - it won't be enough as `js-waku` node won't drop connection to peers so that filter subscription will just continue working;

Maybe I am missing something but I don't see a way to test `navigator.onLine` changes, but I believe it is good enough as this is browser API which just works and the next layer on top is `connection_manager.isConnected()` - which is thoroughly tested already.